File: projects/spring-boot-rest/demo/src/main/java/com/sourabh/demo/controller/RoomController.java
package com.sourabh.demo.controller;

import com.sourabh.demo.model.Room;
import com.sourabh.demo.service.RoomService;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api")
public class RoomController {
    @Autowired
    RoomService roomService;

    @PostMapping("hotels/add/rooms")
    @ApiOperation(value = "Add room to the repository",
            notes = "Provide a Room details and it will be updated in repository!")
    public Room addOrUpdateRoom(@RequestBody Room room){
        return roomService.addOrUpdate(room);
    }


    @GetMapping("hotels/get/rooms")
    @ApiOperation(value = "Get all Rooms",
            notes = "Displays all the rooms available in Hotel.")
    public List<Room> getAll(){
        return roomService.getAll();
    }

    @PostMapping("hotels/add-bulk/rooms")
    @ApiOperation(value = "Add all Rooms",
            notes = "Provide a list of rooms to be added to repository.")
    public List<Room> addAll(@RequestBody List<Room> rooms){
        return roomService.addAll(rooms);
    }

    @DeleteMapping("hotels/del/rooms/{id}")
    @ApiOperation(value = "Flag a room so that it cannot be booked.",
            notes = "Provide an id. based on which if available the room will be not allowed\n" +
                    "to be booked.")
    public Room deleteRoom(@PathVariable int id){
        return roomService.removeRoom(id);
    }

}

File: projects/spring-boot-rest-kt/hotel-reservation/src/main/kotlin/com/sourabh/hotelreservation/controller/RoomController.kt
package com.sourabh.hotelreservation.controller

import com.sourabh.hotelreservation.model.Room
import com.sourabh.hotelreservation.service.RoomService
import io.swagger.annotations.ApiOperation
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api")
class RoomController(@Autowired private val roomService: RoomService) {

    @PostMapping("/rooms")
    @ApiOperation(value = "Add room to the repository", notes = "Provide a Room details and it will be updated in repository!")
    fun addOrUpdateRoom(@RequestBody room: Room):Room{
        return roomService.addOrUpdate(room)
    }

    @GetMapping("/rooms")
    @ApiOperation(value = "Get All Rooms", notes = "Display all rooms available in Hotel.")
    fun getAll():MutableList<Room>?{
        return roomService.getAll()
    }

    @PostMapping("/add-bulk/rooms")
    @ApiOperation(value = "Add all Rooms", notes = "Provide a list of rooms to be added to repository.")
    fun addAll(@RequestBody rooms: MutableList<Room>):MutableList<Room>{
        return roomService.addAll(rooms)
    }

    @DeleteMapping("/rooms{id}")
    @ApiOperation(value = "Flag a room so that it cannot be booked.", notes = """Provide an id. based on which if available the room will be not allowed
to be booked.""")
    fun deleteRoom(@PathVariable id: Int):Room{
        return roomService.removeRoom(id)
    }

}